### PR TITLE
[CodeHealth] Remove `CHECK(false)` uses

### PR DESCRIPTION
**Pull request by cdesouza-chromium**
All uses of `CHECK(false)` are incorrect and should be either a `NOTREACHED`, `LOG(ERROR)`, or higher `CHECK` that removed the enclosing conditional code path.

Resolves https://github.com/brave/brave-browser/issues/42331

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_news/browser/feed_generation_info.cc [hunk 1]
```diff
@@ -285,12 +285,10 @@ mojom::FeedItemMetadataPtr FeedGenerationInfo::PickAndConsume(
...
-  if (index >= articles.size()) {
-    DCHECK(false) << "|index| should never be outside the bounds of |articles| "
-                     "(index: "
-                  << index << ", articles.size(): " << articles.size();
-    return nullptr;
-  }
+  CHECK_LT(index, articles.size())
+      << "|index| should never be outside the bounds of |articles| "
+         "(index: "
+      << index << ", articles.size(): " << articles.size();
```
Comment (goodov) [inline components/brave_news/browser/feed_generation_info.cc:291]: nit: I don't think this detailed explanation is needed here. `CHECK_LT(index, articles.size());` should be enough.

File: components/brave_rewards/core/database/database_util.cc [hunk 2]
```diff
@@ -114,11 +114,7 @@ int GetIntColumn(mojom::DBRecord* record, const int index) {
     return 0;
   }
 
-  if (!record->fields.at(index)->is_int_value()) {
-    DCHECK(false);
-    return 0;
-  }
-
+  DCHECK(record->fields.at(index)->is_int_value());
   return record->fields.at(index)->get_int_value();
```
Comment (goodov) [inline components/brave_rewards/core/database/database_util.cc:118]: all getters has `CHECK` call internally:
```
  int32_t get_int_value() const {
    CHECK(tag_ == Tag::kIntValue);
    return data_.int_value;
  }
```

File: components/brave_rewards/core/database/database_util.cc [hunk 3]
```diff
@@ -114,11 +114,7 @@ int GetIntColumn(mojom::DBRecord* record, const int index) {
     return 0;
   }
 
-  if (!record->fields.at(index)->is_int_value()) {
-    DCHECK(false);
-    return 0;
```
Comment (goodov) [inline components/brave_rewards/core/database/database_util.cc:119]: not sure if these conditions should be removed, but it's up to the wallet team.

**Review by goodov — APPROVED** [review]

**Review by AlexeyBarabash — APPROVED** [review]
sync ++

File: components/brave_rewards/core/database/database_util.cc (hunk 3 above)
Comment (cdesouza-chromium) [reply components/brave_rewards/core/database/database_util.cc:119]: Well, I'm not sure either, but if we must support failure here, then the DCHECK has to go.

File: components/brave_news/browser/feed_generation_info.cc (hunk 1 above)
Comment (cdesouza-chromium) [reply components/brave_news/browser/feed_generation_info.cc]: Fixed.

File: components/brave_rewards/core/database/database_util.cc (hunk 2 above)
Comment (cdesouza-chromium) [reply components/brave_rewards/core/database/database_util.cc]: Fixed.

File: components/brave_rewards/core/database/database_util.cc (hunk 3 above)
Comment (cdesouza-chromium) [reply components/brave_rewards/core/database/database_util.cc:119]: I've reached to @zenparsing and he advised to keep the return for the default value, when there's a mismatch. I have just dropped `DCHECK(false)` from the code.

**Review by zenparsing — APPROVED** [review]
Rewards 👍

**Review by nuo-xu — APPROVED** [review]

**Pull request merged**